external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaUtils.java

```diff
@@ -213,7 +213,11 @@ public static ByteBufferMessageSet fetchMessages(KafkaConfig config, SimpleConsu
                 throw new TopicOffsetOutOfRangeException(msg);
             } else {
                 String message = "Error fetching data from [" + partition + "] for topic [" + topic + "]: [" + error + "]";
-                LOG.error(message);
+                if (error.equals(KafkaError.NOT_LEADER_FOR_PARTITION)) {
```

Comment:
I see. Then we can remove this log itself since the same message is being logged again at https://github.com/apache/storm/blob/master/external/storm-kafka/src/jvm/org/apache/storm/kafka/KafkaSpout.java#L143. 
TridentKafkaSpout is swallowing up the exception. You can add the exception in the log there.